### PR TITLE
Fix latest CANN bugs (repeat ops) on OpenSora1.1

### PR DESCRIPTION
**Pull request by SamitHuang**
# What does this PR do?

Fixes # (issue)
hpcai opensora v1.1 cannot run on latest CANN version (0630, 0530, or later), due to repeat / tile ops are not supported in uint8 anymore

Adds # (feature)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/opensora_hpcai/scripts/args_train.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def parse_train_args(parser):
         help="Specify the [beta1, beta2] parameter for the AdamW optimizer.",
     )
     parser.add_argument(
-        "--optim_eps", type=float, default=1e-6, help="Specify the eps parameter for the AdamW optimizer."
+        "--optim_eps", type=float, default=1e-8, help="Specify the eps parameter for the AdamW optimizer."
```
Comment (zhanghuiyao) [inline examples/opensora_hpcai/scripts/args_train.py:114]: 修改eps为1e-8的表现可能不一样，建议非必要不改

File: mindone/trainers/adamw_mf.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                self._parameters,
+                gradients,
+                self.exp_avg,
+                self.exp_avg_sq,
+                self.optim_filter,
+            )
+        if self.use_parallel:
+            self.broadcast_params(optim_result)
+
+        return optim_result
```
Comment (zhanghuiyao) [inline mindone/trainers/adamw_mf.py:269]: 这个是否用的是mindformer的版本，精度是否能跟torch.adamw 对齐

**Review by zhanghuiyao — APPROVED** [review]

File: examples/opensora_hpcai/scripts/args_train.py (hunk 1 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/scripts/args_train.py:114]: config里也是手写设置1e-8

File: mindone/trainers/adamw_mf.py (hunk 2 above)
Comment (SamitHuang) [reply mindone/trainers/adamw_mf.py:269]: mindformer的版本性能略慢一些, ~100ms. 作为一个option

**Review by hadipash — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

File: examples/opensora_hpcai/opensora/pipelines/train_pipeline.py [hunk 3]
```diff
@@ -352,7 +352,8 @@ def compute_loss(
 
         # Learn the variance using the variational bound, but don't let it affect our mean prediction.
         patch_mask = temporal_mask[:, :, None, None] * spatial_mask[:, None, :, None]
-        patch_mask = self.unpatchify(ops.tile(patch_mask, (1, 1, 1, D)))  # b c t h w
+        pm_dtype = patch_mask.dtype
+        patch_mask = self.unpatchify(ops.tile(patch_mask.to(ms.int32), (1, 1, 1, D)).to(pm_dtype))  # b c t h w
```
Comment (zhtmike) [inline examples/opensora_hpcai/opensora/pipelines/train_pipeline.py:356]: as a temporary fix due to MS/CANN bug, maybe not merge to master?

Comment (SamitHuang) [reply examples/opensora_hpcai/opensora/pipelines/train_pipeline.py:356]: the performance of ops.tile / repeat in uint8 is no better than that in int32, so it's worthy.

Comment (zhtmike) [reply examples/opensora_hpcai/opensora/pipelines/train_pipeline.py:356]: though the time for uint8/int32 is same, casting to int32 and casting back will add two extra casting operations, may not be nesessary

**Review by vigo999 — APPROVED** [review]

**Review by HaoyangLee — APPROVED** [review]
Get the same training speed on cloud as on local server with latest CANN.

**Pull request merged**